src/alpaka/AlpakaCore/alpakaWorkDivHelper.h

```diff
@@ -8,234 +8,225 @@ using namespace alpaka_common;
 #else
       // On the CPU:
       // Run serially with a single thread per block: threadsPerBlock is always 1.
       // threadsPerBlockOrElementsPerThread is the number of elements per thread.
-      const Vec<T_Dim>& threadsPerBlock = Vec<T_Dim>::ones();
-      return WorkDiv<T_Dim>(blocksPerGrid, threadsPerBlock, threadsPerBlockOrElementsPerThread);
+      const Vec<TDim>& threadsPerBlock = Vec<TDim>::ones();
+      return WorkDiv<TDim>(blocksPerGrid, threadsPerBlock, threadsPerBlockOrElementsPerThread);
 #endif
     }
 
-    /*
-     * 1D helper to only access 1 element per block 
-     * (should obviously only be needed for debug / printout).
-     */
-    template <typename T_Acc>
-    ALPAKA_FN_ACC bool once_per_block_1D(const T_Acc& acc, uint32_t i) {
-      const uint32_t blockDimension(alpaka::getWorkDiv<alpaka::Block, alpaka::Elems>(acc)[0u]);
-      return (i % blockDimension == 0);
-    }
+    /*********************************************
+     *           RANGE COMPUTATION
+     ********************************************/
 
     /*
      * Computes the range of the elements indexes, local to the block.
      * Warning: the max index is not truncated by the max number of elements of interest.
      */
-    template <typename T_Acc, typename T_Dim = alpaka::Dim<T_Acc>>
-    ALPAKA_FN_ACC std::pair<Vec<T_Dim>, Vec<T_Dim>> element_index_range_in_block(const T_Acc& acc,
-                                                                                 const Vec<T_Dim>& elementIdxShift) {
-      Vec<T_Dim> firstElementIdxVec = Vec<T_Dim>::zeros();
-      Vec<T_Dim> endElementIdxUncutVec = Vec<T_Dim>::zeros();
-
-      // Loop on all grid dimensions.
-      for (typename T_Dim::value_type dimIndex(0); dimIndex < T_Dim::value; ++dimIndex) {
-        // Take into account the thread index in block.
-        const uint32_t threadIdxLocal(alpaka::getIdx<alpaka::Block, alpaka::Threads>(acc)[dimIndex]);
-        const uint32_t threadDimension(alpaka::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc)[dimIndex]);
-
-        // Compute the elements indexes in block.
-        // Obviously relevant for CPU only.
-        // For GPU, threadDimension = 1, and elementIdx = firstElementIdx = threadIdx + elementIdxShift.
-        const uint32_t firstElementIdxLocal = threadIdxLocal * threadDimension;
-        const uint32_t firstElementIdx = firstElementIdxLocal + elementIdxShift[dimIndex];  // Add the shift!
-        const uint32_t endElementIdxUncut = firstElementIdx + threadDimension;
-
-        firstElementIdxVec[dimIndex] = firstElementIdx;
-        endElementIdxUncutVec[dimIndex] = endElementIdxUncut;
-      }
+    template <typename TAcc>
+    ALPAKA_FN_ACC std::pair<Idx, Idx> element_index_range_in_block(const TAcc& acc,
+                                                                   const Idx elementIdxShift,
+                                                                   const Idx dimIndex = 0u) {
```

Comment:
If I read the code correctly, this is the index of the dimension (i.e. the equivalent of `.x`, `.y`, or `.z` in CUDA).
In this case the type should probably not be `Idx` but `unsigned int` ?

Reply:
Yes, correct. At first it was `alpaka::Dim<TAcc>`, which is not possible anymore as, like explained below, a simple 1D access is used also for 2D grids (kernels launched with `Acc2`), for each dimension. Using `Dim1` (`alpaka::DimInt<1u>`) would be very inelegant, because of the use for 2D grids. Hence I have just used `Idx.` 
But indeed, having simply `unsigned int` (or better, a `DimIdx` ?) is the way to go, will add a small correction (also accordingly on call site).

Reply:
Just replaced `Idx` for dim indices with `unsigned int` in the end, likely indices types and their use on call sites would need to be thought of more though.

Reply:
To be consistent, do you think `Idx` should be used on call sites instead of `uin32_t` for element indices? (even if that's obviously the same right now).